### ISSUE TITLE
VPN service issue - ZenMate

### ISSUE THREAD

**Issue by evgeniy-ADG** [AdguardTeam/AdguardForAndroid#1044]
Cannot enable Adguard, the only solution is to uninstall ZenMate

STR:

1. Install ZenMate ([link](https://play.google.com/store/apps/details?id=com.zenmate.android&hl=en))
2. Launch Adguard
3. Launch ZenMate when Adguard is running
4. It interrupts Adguard and establishes its own VPN
5. Try to launch Adguard then, you'll see that it's impossible - ZenMate continues interrupting Adguard unless you uninstall it

If you go to ZenMate and stop it before starting Adg, you'll face the same problem - it won't start.
Clearing caches and data for both apps didn't help, too.

Reproduced on Samsung A5 2016. Not sure if it's specific to Samsung devices only.

Ticket: 1498940

**Comment by vozersky** [comment]
@evgeniy-ADG closing ZenMate app should be enough actually. If not, try force stopping it.

**Comment by evgeniy-ADG** [comment]
It should be but it's not... Force stopping did the trick.

**Comment by vozersky** [comment]
@evgeniy-ADG 

The problem is that ZenMate doesn't let other apps to use VPN. I will have to report it to ZenMate devs. 

### Steps to reproduce
1. Install, open and login to ZenMate app (You don't even have to start protection).
2. Install and open any other VPN app. (The issue affects literally any VPN-based app) 
    For example: Hotspot Shield (over 10M installs): https://play.google.com/store/apps/details?id=hotspotshield.android.vpn
2. Open app, skip trial section and press "Connect" (which establishes a VPN connection)

### Expected result

VPN is up and running.
### Actual result

VPN is stopped shortly after turning ON.

### Analysis

It seems like ZenMate revokes VPN access rights from other apps.
The thing is that on Android VPN apps can easily mess with each other one by calling `VpnService.prepare` method. In order to fix this issue they could check other VPN presence before doing this call.

### Possible solution 
To check for "tun" network interface presence before calling VpnService.prepare:
https://gist.github.com/ameshkov/5d8bb32f36b915d303be519eaff82538﻿


Here are some popular apps affected by this bug:
[Hola (over 10M installs)](https://play.google.com/store/apps/details?id=org.hola)
[HotSpot Shield (over 10M installs)](https://play.google.com/store/apps/details?id=hssb.android.free.app)
[OpenVPN Connect (over 10M installs)](https://play.google.com/store/apps/details?id=net.openvpn.openvpn)
[TunnelBear(Over 1M installs)](https://play.google.com/store/apps/details?id=com.tunnelbear.android)

**Comment by ameshkov** [comment]
@vozersky let's open a new issue "VPN apps compatibility" and keep track of the apps affected by this issue there.

We'll have to report this issue to the developers of every affected app.

**Comment by vozersky** [comment]
#1046

**Comment by veronikasav** [comment]
Does ZenMate VPN extension compatible with Adguard program?

**Comment by ameshkov** [comment]
@veronikasav 
> Does ZenMate VPN extension compatible with Adguard program?

Are you talking about Android or Windows?

**Comment by veronikasav** [comment]
Sorry, the extension for Opera on Windows.

**Comment by evgeniy-ADG** [comment]
she's talking about [this ](https://addons.opera.com/ru/extensions/details/zenmate-for-operatm/?display=ru), just checked it, seems to work fine with adguard,
